applications/nrf5340_audio/src/audio_module/src/amod_api.h

```diff
@@ -0,0 +1,492 @@
+			      size_t stack_size, int priority, struct data_fifo *msg_rx,
+			      struct data_fifo *msg_tx, struct k_mem_slab *data_slab,
+			      size_t data_size);
+
+/**
+ * @brief Helper function to return the base and instance names for a given
+ *        module handle.
+ *
+ * @param handle         A handle to this module instance
+ * @param base_name      A pointer to the name of the module
+ * @param instance_name  A pointer to the name of the current module instance
+ *
+ * @return 0 if successful, error value
+ */
+int amod_names_get(struct amod_handle *handle, char *base_name, char *instance_name);
+
+/**
+ * @brief Helper function to return the state of a given module handle.
+ *
+ * @param handle  A handle to this module instance
+ *
+ * @return 0 if successful, error value
+ */
+int amod_state_get(struct amod_handle *handle);
+
+/**
+ * @brief Helper function to attach a raw audio data buffer to an audio block.
+ *
+ * @param block        Pointer to the audio data block to fill
+ * @param data_type    The type of data carried in the block
+ * @param data         Pointer to the raw or coded audio data buffer
+ * @param data_size    Size of the raw or coded audio data buffer
+ * @param format       The format od the data carried in the block
+ * @param reference_ts Reference timestamp to be used to synchronise the data
+ * @param block_rx_ts  Block received timestamp to be used to synchronise the data
+ * @param bad_frame    A flag to indicate there are errors within the data for this block
+ * @param last_flag    A flag to indicate this is the last block in the stream
+ * @param user_data    A pointer to a private area of user data or NULL
+ *
+ * @return 0 if successful, error value
+ */
+int amod_block_data_attach(struct aobj_block *block, enum aobj_type data_type, char *data,
+			   size_t data_size, struct aobj_format *format, uint32_t *reference_ts,
+			   uint32_t *block_rx_ts, bool bad_frame, bool last_flag, void *user_data);
+
+/**
+ * @brief Helper function to extract the raw audio data buffer from an audio block.
+ *
+ * @param block        Pointer to the audio data block to fill
+ * @param data_type    The type of data carried in the block
+ * @param data         Pointer to the raw or coded audio data buffer
+ * @param data_size    Size of the raw or coded audio data buffer
+ * @param format       The format od the data carried in the block
+ * @param reference_ts Reference timestamp to be used to synchronise the data
+ * @param block_rx_ts  Block received timestamp to be used to synchronise the data
+ * @param bad_frame    A flag to indicate there are errors within the data for this block
+ * @param last_flag    A flag to indicate this is the last block in the stream
+ * @param user_data    A pointer to a private area of user data or NULL
+ *
+ * @return 0 if successful, error value
+ */
+int amod_block_data_extract(struct aobj_block *block, enum aobj_type *data_type, char *data,
```

Comment:
Since the calling function already has to include aobj_api.h to be able to get aobj_block, why should the user use this function to extract the data when they can just read it directly from the struct?

Reply:
See above.